workspaces/src/operations.rs

```diff
@@ -24,17 +24,21 @@ const MAX_GAS: Gas = 300_000_000_000_000;
 /// A set of arguments we can provide to a transaction, containing
 /// the function name, arguments, the amount of gas to use and deposit.
 #[derive(Debug)]
-pub struct Function<'a> {
-    name: &'a str,
-    args: Result<Vec<u8>>,
-    deposit: Balance,
-    gas: Gas,
+pub struct FunctionArgs<T> {
+    pub(crate) name: T,
+    pub(crate) args: Result<Vec<u8>>,
+    pub(crate) deposit: Balance,
+    pub(crate) gas: Gas,
 }
 
-impl<'a> Function<'a> {
+pub type Function<'a> = FunctionArgs<&'a str>;
+/// Owned version of [`Function`], whereby a lifetime is not attached.
+pub type FunctionOwned = FunctionArgs<String>;
```

Comment:
It doesn't look like the non-owned type pulls its weight. `Function` will eventually lead to a network call, a string allocation is nothing compared to that.

Reply:
That's true. `Function<'a>` is leaning towards micro optimization, but also makes the user experience a bit rougher if they wanted to store these `Function` objects in their own struct. I'll just move to the owned version unless there was something else. @austinabell if you had any opinions about this